File: LB2/Services/RsaService.cs
// using System.Security.Cryptography;
//
// namespace LB2.Services;
//
// public interface IRsaService
// {
//     string PublicKey { get; }
//     string PrivateKey { get; }
//     void GenerateKeys();
// }
//
// public class RsaService : IRsaService
// {
//     public string PublicKey { get; private set; }
//     public string PrivateKey { get; private set; }
//
//     public void GenerateKeys()
//     {
//         using var rsa = new RSACryptoServiceProvider(2048);
//         PublicKey = rsa.ExportSubjectPublicKeyInfoPem();
//         PrivateKey = rsa.ExportSubjectPublicKeyInfoPem();
//     }
// }

File: clients/sopad/src/app/services/rsa.service.ts
import {Injectable} from '@angular/core';
import * as forge from 'node-forge'
import {environment} from "../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class RsaService {

  privateKey: string = ''
  publicKey: string = ''

  constructor() {
  }

  generateKeys() {
    const keyPair = forge.pki.rsa.generateKeyPair({bits: 2048})
    this.publicKey = forge.pki.publicKeyToPem(keyPair.publicKey)
    this.privateKey = forge.pki.privateKeyToPem(keyPair.privateKey)

    const data = 'fasdfasdf'
    const aa = forge.util.encode64(keyPair.publicKey.encrypt(data, 'RSA-OAEP'))
    const bb = keyPair.privateKey.decrypt(forge.util.decode64(aa), 'RSA-OAEP')
    console.log(bb)
  }
}
